proposals/NNNN-byte-address-buffer-alignment.md

```diff
@@ -0,0 +1,983 @@
+                                                    // Note: uint4 largest scalar type is uint (4 bytes)
+uint data2 = MyBuffer.Load<uint>(offset2);         // DXC: min(64, 4) = 4 → DXIL alignment = 4
+uint64_t data3 = MyBuffer.Load<uint64_t>(offset3); // DXC: min(64, 8) = 8 → DXIL alignment = 8
+
+// Example where largest scalar type size exceeds BaseAlignment:
+[BaseAlignment(4)]  // Minimum allowed BaseAlignment
+RWByteAddressBuffer MyOtherBuffer : register(u1);
+
+uint64_t data4 = MyOtherBuffer.Store<uint64_t>(offset4, data3); // DXC error: BaseAlignment = 4 < scalar type alignment = 8
+```
+
+> **Implementation note**: The `BaseAlignment` attribute affects both the base address alignment information in DXIL
+> resource properties and the operation-level alignment calculation for existing `Load`/`Store` functions. This ensures
+> consistent alignment semantics across all buffer access methods. Code that doesn't use `BaseAlignment` continues to
+> work unchanged. Code that adds `BaseAlignment` may experience improved alignment for current operations.
+
+#### Behavior of AlignedLoad/AlignedStore Functions without BaseAlignment Attribute
+
+When the `BaseAlignment` attribute is _not_ specified on the buffer, but the new `AlignedLoad`/`AlignedStore` functions
+are used, then DXC uses the existing alignment requirements to calculate the effective alignment:
+
+> **HLSL Specification, 1.7.2 Memory Spaces**: _"The alignment requirements of an offset into device memory space is the
+> size in bytes of the largest scalar type contained in the given aggregate type"_
+
+If the `alignment` parameter passed to the `AlignedLoad`/`AlignedStore` functions is smaller than the scalar alignment,
+then the compiler will issue an error message.  If the value is larger, then the compiler will issue a warning message
+to inform the developer that this mismatch may be unexpected.  However, for the sake of code reuse, this mismatch will
+be allowed.
+
+#### BaseAlignment Attribute on Function Parameters
+
+The `BaseAlignment` attribute can be applied to `[RW]ByteAddressBuffer` function parameters to specify alignment
+requirements during parameter passing. This enables functions to declare their alignment assumptions explicitly and
+ensures that callers provide buffers with sufficient alignment guarantees. The attribute follows standard alignment
+decay rules where parameter alignment can be less than or equal to the argument's declared alignment, but cannot exceed
+it.
+
+When a function parameter specifies `BaseAlignment`, the argument passed to that function must have been declared with
+`BaseAlignment`, and the argument's alignment value must be greater than or equal to the parameter's alignment
+requirement. If the parameter does not specify `BaseAlignment`, arguments with or without `BaseAlignment` can be passed.
+The effective alignment calculation within the function uses the parameter's declared `BaseAlignment` value, not the
```

Comment:
By "calculation", I imagine you mean for the purpose of illegal alignment widening diagnostics in Sema, rather than something that impacts the generated DXIL?